### PR TITLE
chore: wait for frames to be destroyed

### PR DESCRIPTION
**Pull request by jrandolf-google**

### CODE REVIEW

File: packages/puppeteer-core/src/bidi/Frame.ts [hunk 1]
```diff
@@ -267,11 +283,7 @@ export class BidiFrame extends Frame {
       this.waitForNavigation(options),
       this.browsingContext.navigate(
         url,
-        // Some implementations currently only report errors when the
-        // readiness=interactive. This also ensures that old frames have been
-        // removed.
-        // Related: https://bugzilla.mozilla.org/show_bug.cgi?id=1846601
-        Bidi.BrowsingContext.ReadinessState.Interactive
+        Bidi.BrowsingContext.ReadinessState.None
```
Comment (OrKoN) [inline packages/puppeteer-core/src/bidi/Frame.ts]: This is not only about frames but also about error reporting.

Comment (jrandolf) [reply packages/puppeteer-core/src/bidi/Frame.ts]: I've updated the comment.

File: packages/puppeteer-core/src/bidi/Frame.ts [hunk 2]
```diff
@@ -258,19 +260,33 @@ export class BidiFrame extends Frame {
     });
   }
 
+  #detached$() {
+    if (this.detached) {
+      return of(this as Frame);
+    }
+    return fromEmitterEvent(
+      this.page().trustedEmitter,
+      PageEvent.FrameDetached
```
Comment (OrKoN) [inline packages/puppeteer-core/src/bidi/Frame.ts]: should this also wait for its own child frames?

Comment (jrandolf) [reply packages/puppeteer-core/src/bidi/Frame.ts]: All the child frames should be destroyed at this point.

**Review by OrKoN — APPROVED** [review]

**Pull request merged**